bin/main.rb

```diff
@@ -13,104 +12,103 @@
+    @player1 = Player.new(player1_name)
+    @player1.type = 'X'
+    @player2 = Player.new(player2_name)
+    @player2.type = 'O'
+    @board = Board.new
   end
-end
-
-def play(player1, player2)
-  board1 = Board.new
-  loop do
-    board1.draw
-    user_input(player1, board1)
-    break unless check_outcome(-1, player1).nil?
 
-    board1.draw
-    user_input(player2, board1)
-    break unless check_outcome(-1, player2).nil?
-  end
-end
+  def play
+    loop do
+      @board.draw
+      input(@player1)
+      system('clear')
+      break unless check_outcome(@player1) == -1
 
-def user_input(player, board1)
-  played = false
-  until played
-    puts "\n#{player.name}'s turn"
-    puts "\nEnter position (eg. A2)"
-    input_pos = gets.chomp
-    system('clear')
-    played = legal(player.type, input_pos, board1)
+      @board.draw
+      input(@player2)
+      system('clear')
+      break unless check_outcome(@player2) == -1
+    end
   end
-end
 
-def update(value, pos, board1)
-  arr = pos.split('')
-  arr[0] = arr[0].upcase
-  case arr[0]
-  when 'A' then arr[0] = 0
-  when 'B' then arr[0] = 1
-  when 'C' then arr[0] = 2
+  def input(player)
+    playing = false
+    until playing
+      puts "\n#{player.name}'s turn"
+      puts "\nEnter position (eg. A2)"
+      input_pos = gets.chomp
+      legal = @board.legal_input?(input_pos)
+      if legal
+        pos = clean_position(input_pos)
+        unless @board.cell_empty?(pos)
+          puts '*** This cell is not empty. Please try again. ***'
+          next
+        end
+        @board.update(player.type, pos)
+        playing = true
+      else
+        puts '*** This move is illegal. Please try again. ***'
+      end
+    end
   end
-  board1.board[arr[0]][arr[1].to_i - 1] = value
-end
 
-def cell_empty?(arr, board1, pos, value)
-  if board1.board[arr[0]][arr[1].to_i - 1] == '_'
-    puts board1.board[arr[0]][arr[1].to_i - 1]
-    update(value, pos, board1)
-    true
-  else
-    puts '*** Please choose an empty cell. ***'
-    false
+  def clean_position(input_pos)
+    arr = input_pos.split('')
+    x_axis = arr[1].to_i - 1
+    y_axis = arr[0].upcase
+    # convert string input to array indices
+    case y_axis
+    when 'A' then y_axis = 0
+    when 'B' then y_axis = 1
+    when 'C' then y_axis = 2
+    end
+    [y_axis, x_axis]
   end
-end
 
-def legal(value, pos, board1)
-  arr = pos.split('')
-  arr[0] = arr[0].upcase
-  if (%w[A B C].include? arr[0]) && (%w[1 2 3].include? arr[1])
-    case arr[0]
-    when 'A' then arr[0] = 0
-    when 'B' then arr[0] = 1
-    when 'C' then arr[0] = 2
+  # Prints out game result
+  def check_outcome(player)
+    case @board.status
+    when 1
+      puts "*** #{player.name} Won Congratulations! ***"
+      1
+    when 0
+      puts "*** That's a tie ***"
+      0
+    else
+      -1
     end
-    cell_empty?(arr, board1, pos, value)
-  else
-    puts '*** This move is illegal please try again. ***'
-    false
   end
 end
 
-def status; end
-
```

Comment:
please move the game class to it's own file inside the lib folder and keep the main file for only running the game and for the `puts` & `gets`